behave/features/steps/public_end_to_end.py

```diff
@@ -0,0 +1,37 @@
+from behave import step
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.keys import Keys
+
+
+@step('I click on the "Exit this page" button')
+def step_impl_click_exit_page(context):
+    context.helperfunc.click_button(By.CLASS_NAME, "govuk-exit-this-page")
+
+
+@step('I press the "{keypress}" key "{amount_of_time}" on the keyboard')
+def step_impl_keypress_multiple_times(context, keypress, amount_of_time):
+    context.helperfunc.find_by_class("govuk-js-exit-this-page-skiplink")
+
+    if (
+        len(amount_of_time) == 3
```

Comment:
My apologies if I am misinterpreting this line, however, `amount_of_time` is being passed into this step implementation as a string, meaning `len()` is getting the length of the string.

Should this function not be implemented as below:
```
@step('I press the "{keypress}" key "{amount_of_time} times" on the keyboard')
def step_impl_keypress_multiple_times(context, keypress, amount_of_time):
    context.helperfunc.find_by_class("govuk-js-exit-this-page-skiplink")

    key_to_press = Keys.SHIFT if keypress == "shift" else Keys.TAB if keypress == "tab" else None

    for _ in range(amount_of_time):
        if (key_to_press):
            ActionChains(webdriver).send_keys(key_to_press).perform()
```

Additionally, there should be an additional step implementation for either pressing 'Enter', or clicking on the 'Exit this page' link after pressing 'Tab' twice.

Many thanks 😄

Reply:
+1